### PR TITLE
Tidying up for packaging

### LINKED ISSUE #2: Update ubuntu-packaging-guide archive packages
The Ubuntu Packaging Guide is currently also distributed as a debian package (see: https://launchpad.net/ubuntu/+source/ubuntu-packaging-guide/).

~~In the last weekly meeting we discussed to drop the support for this form of distribution as it does not provide a lot of value in the presence of a Read the Docs version.~~

This issue should track this effort.

### PR DESCRIPTION
**Pull request by waveform80**
This is a whole pile of commits which seem to do a lot but should (hopefully) be entirely transparent to anyone actually using the docs. Specifically:

* There are several changes to the Makefile structure; some targets are added which are required by the packaging (e.g. distclean), others are purely for convenience (to ensure a bare "make" invocation from the root offers the help text correctly)
* The venv handling is changed slightly (not from the user perspective, but just implementation) to ensure it can be overridden entirely by the packaging system (which must not use venvs when building as that would require internet access).
* The sphinx requirements are lowered to 4.3.2. This won't affect anyone using the venv, but I wanted to check it would work so that eventually we could target jammy for a backport of the packaging; it works happily so I saw no harm in leaving it there.
* The EPUB and PDF output is cleaned up a little, to include proper link references, authorship, and copyright.
* The output structure is switched from `dirhtml` to `html`. This changes a few URLs (e.g. `search/index.html` becomes `search.html`) but firstly matches what we use on ReadTheDocs anyway, and secondly is required for the debhelper packaging system.
* Finally, different formats are now built in different sub-dirs of `docs/_build` which keeps things simple when working with multiple formats (and also simplifies the packaging task greatly).

### CODE REVIEW

File: docs/conf.py
Comment (s-makin) [inline docs/conf.py:37]: I would prefer to keep m2r2 here if possible. It will give us flexibility to accommodate people writing in both markdown and reST if we can convert files submitted in md to rst.

Comment (dviererbe) [reply docs/conf.py:37]: I am not keen of having 2 formats, instead we should encourage people to submit articles as markdown and I (and if I recall correctly Dave too) am happy to translate it to reST.

File: Makefile
Comment (dviererbe) [inline Makefile:1]: Praise: Did not know about `$(MAKE)` https://www.gnu.org/software/make/manual/make.html#MAKE-Variable

File: docs/Makefile
Comment (dviererbe) [inline docs/Makefile:23]: todo: make target `distclean` is added, but not documented in the `help` target. 

Maybe we can add something like this:
```suggestion
	@echo "* clean     -- clean full environment"
	@echo "* distclean -- clean full environment and built doc files"
```

File: docs/Makefile
Comment (dviererbe) [inline docs/Makefile:59]: suggestion: reuse `clean-doc` and `clean` target

```suggestion
distclean: clean-doc clean
```

File: docs/Makefile
Comment (dviererbe) [inline docs/Makefile:31]: thought: Should we use a `$(PYTHON)` variable?

**Review by dviererbe — CHANGES_REQUESTED** [review]
Thanks @waveform80, approve when `distclean` make target is documented.

The other requested changes are optional.

File: docs/conf.py
Comment (s-makin) [reply docs/conf.py:37]: m2r2 would do that automatically, but if you're both happy to do it by hand then there's no issue with removing the extension and this can be marked resolved

Comment (waveform80) [reply docs/conf.py:37]: The other issue with keeping `m2r2` is it's not packaged (in any release of Ubuntu); I can add a quilt patch to my packaging branch to excise the requirement when building the package but I quite liked that (currently) there are no patches at all in the packaging branch.

Still, while @dviererbe is absolutely correct that I'm happy to translate anything to reST, I'll defer to the wisdom of the documentation team here and remove the ... erm ... removal!

File: docs/Makefile
Comment (waveform80) [reply docs/Makefile:23]: Yes, that was slightly deliberate as I only added it for the convenience of the packaging branch (i.e. it's not something I really intend end-users to know about or use given the `clean` and `clean-doc` targets already existed). I'm happy to document it if required, but we might want to keep it "semi-hidden" for the reasons given below...

File: docs/Makefile
Comment (waveform80) [reply docs/Makefile:59]: Unfortunately this is not really possible; the `clean-doc` target uses `git clean` but when packaging it's not dealing with a git repo, but an unpacked tar-ball so this cannot work.

File: docs/Makefile
Comment (waveform80) [reply docs/Makefile:31]: Hmm, that's quite a neat idea just in case someone wants to use a non-system Python (given everything then occurs in a venv, that's an entirely valid way of working). I'll add it.

File: docs/Makefile
Comment (dviererbe) [reply docs/Makefile:59]: Oh... For some reason I did not register that `clean-doc` has a completely different command. Never mind.

File: docs/Makefile
Comment (dviererbe) [reply docs/Makefile:23]: That makes sense.

File: docs/Makefile
Comment (dviererbe) [reply docs/Makefile]: @waveform80 I added it here: https://github.com/canonical/ubuntu-packaging-guide/pull/16/commits/8c1d61ff90329101b86d2e8fe68f249b35132515

If this is okay, feel free to merge.

**Review by dviererbe — APPROVED** [review]

**Pull request merged**
